lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -688,6 +696,18 @@ class SqlParser(private val ion: IonSystem) : Parser {
                 val (year, month, day) = dateString.split("-")
                 DateTimeType.Date(year.toInt(), month.toInt(), day.toInt(), metas)
             }
+            TIME -> {
+                val timeString = token!!.text!!
+                val precision = min(children[0].token?.value?.numberValue()?.toInt() ?: DEFAULT_PRECISION_FOR_TIME, DEFAULT_PRECISION_FOR_TIME)
```

Comment:
Use of `min` here seems like a bad idea.  Why shouldn't this be an error if the user specifies a precision that is greater than what is supported?  (Postgres does this for precisions beyond 6, BTW.)

Reply:
So I checked with the `4.21` version of `pgAdmin 4` and it allows specifying the precision beyond the maximum. 
For example, this query succeeds. 
`SELECT TIME (19) '12:23:30.1234567898'` -> `12:23:30.123457`